packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ProxyMethodMapper.java

```diff
@@ -68,397 +57,136 @@ public static ProxyMethodMapper getInstance() {
-                requestContentType = request.getProtocol().getHttp().getMediaTypes().get(0);
-            } else if (request.getProtocol().getHttp().getKnownMediaType() != null) {
-                requestContentType = request.getProtocol().getHttp().getKnownMediaType().getContentType();
-            }
-            builder.requestContentType(requestContentType);
+            final ProxyMethod.Builder builder = baseMethod.newBuilder();
             builder.baseURL(request.getProtocol().getHttp().getUri());
             builder.urlPath(request.getProtocol().getHttp().getPath());
             builder.httpMethod(HttpMethod.valueOf(request.getProtocol().getHttp().getMethod().toUpperCase()));
+            final String contentType = getRequestContentType(request);
+            builder.requestContentType(contentType);
 
-            List<ProxyMethodParameter> parameters = new ArrayList<>();
-            List<ProxyMethodParameter> allParameters = new ArrayList<>();
-            List<ProxyMethod> proxyMethods = new ArrayList<>();
-            // add content-type parameter to allParameters when body is optional and there is single content type
-            if (settings.isDataPlaneClient()
-                // only if "content-type" is not already defined in parameters
-                && request.getParameters()
-                    .stream()
-                    .noneMatch(p -> p.getProtocol() != null
-                        && p.getProtocol().getHttp() != null
-                        && p.getProtocol().getHttp().getIn() == RequestParameterLocation.HEADER
-                        && "content-type".equalsIgnoreCase(p.getLanguage().getDefault().getSerializedName()))) {
-                boolean isBodyParamRequired = request.getParameters()
-                    .stream()
-                    .filter(p -> p.getProtocol() != null
-                        && p.getProtocol().getHttp() != null
-                        && p.getProtocol().getHttp().getIn() == RequestParameterLocation.BODY)
-                    .map(Parameter::isRequired)
-                    .findFirst()
-                    .orElse(false);
-                if (MethodUtil.getContentTypeCount(operation.getRequests()) == 1 && !isBodyParamRequired) {
-                    Parameter contentTypeParameter = MethodUtil.createContentTypeParameter(request, operation);
-                    allParameters.add(Mappers.getProxyParameterMapper().map(contentTypeParameter));
-                }
-            }
+            final ProxyMethodParameterProcessor.Result r = parameterProcessor.process(request, contentType);
+            builder.parameters(r.parameters);
+            builder.allParameters(r.allParameters);
+            builder.specialHeaders(r.specialHeaderParameterNames);
+            builder.name(methodNameGenerator.getUniqueName(r.parameters.stream(), contentType));
+            builder.examples(getExamples(operation, baseMethod.getOperationId()));
```

Comment:
nit, since they are for base(normal) proxy method, maybe put them into `createProxyMethods` as well?

Reply:
yes, good point, I moved the logic to create base method to `createProxyMethods`